request.go

```diff
@@ -124,3 +128,40 @@ func detectEncoding(opts RequestOptions) encoding.Encoding {
 
 	return encoding.JSON
 }
+
+// prepares the request by injecting metadata, applying plugin-based transport middleware,
+// before finally adding any user-provided override values
+func prepareRequest(req *transport.Request, headers map[string]string, opts Options) (*transport.Request, error) {
+	// Add request metadata
+	req.TargetService = opts.TOpts.ServiceName
+
+	// Apply middleware
```

Comment:
why do we apply middleware first? does it make more sense to apply middleware at the very end?

e.g., when you use middleware in other frameworks, you have a request with a body, baggage, headers, and the middleware can then modify it

Reply:
The idea was to apply the middleware first, such that it could be overridden by the CLI options afterwards.

Another option would be to create the fully-formed request first, but ask that `RequestInterceptor` implementations do not override pre-existing values, if they wish to take advantage of them.